### ISSUE TITLE
combine an existing YOLO dataset with a custom dataset

### ISSUE THREAD

**Issue by mazik1496** [ultralytics/ultralytics#1477]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

![image](https://user-images.githubusercontent.com/80936392/226088308-11de79c9-41b0-474d-bd10-884b46fda028.png)

We are merging because the existing yolo dataset and the custom dataset can be merged. It seems to be merged, but if you look at the learning results, you need to recognize a person, but the class of the custom dataset is recognized as above. May I know the cause and effect for this? 
In the case of merging, the images and labels of the existing yolo dataset were moved to the custom data, and then the yaml file of the custom data was modified and training was performed.

**Comment by mazik1496** [comment]
As a result of searching for the current problem, the total number of classes in the existing dataset and custom dataset is 83, and it is clear that 2-3 of them are missing. I will add attachments and open sources. Thank you for watching.

-------------source code---------------
#!/usr/bin/env python3

from ultralytics import YOLO

# Load a model
model = YOLO("yolov8n.yaml")  # build a new model from scratch
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)

# Use the model
model.train(data='/home/user/vrx_ws/src/vrx/vrx_gazebo/nodes/data.yaml', epochs=3)  # train the model
metrics = model.val()  # evaluate model performance on the validation set
results = model("/home/user/vrx_ws/src/vrx/vrx_gazebo/nodes/Result_of_Original.png")  # predict on an image
success = model.export(format="onnx")  # export the model to ONNX format


-----------------------yaml code----------------------------------
train: /home/user/vrx_ws/src/vrx/vrx_gazebo/nodes/train/images
val: /home/user/vrx_ws/src/vrx/vrx_gazebo/nodes/valid/images
test: /home/user/vrx_ws/src/vrx/vrx_gazebo/nodes/test/images

nc: 83
names: ['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck',
        'boat', 'traffic light', 'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse',
        'sheep', 'cow', 'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase',
        'frisbee', 'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
        'tennis racket', 'bottle', wine glass', 'cup', 'fork', 'knite', 'spoon', 'bowl', 'banana', 'apple', 'sandwich',
        'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed',
        'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'ceil phone', 'microwave', 'oven',
        'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear', 'hair drier', 'toothbrush','Bicon', 'dock', 'light_buoy']

roboflow:
  workspace: project
  project: vrx-detection
  version: 1
  license: CC BY 4.0
  url: https://universe.roboflow.com/project/vrx-detection/dataset/1

**Comment by pderrenger** [comment]
@mazik1496 it seems like the issue might be related to the class mapping between the existing YOLO dataset and the custom dataset. When combining datasets, it's essential to ensure that the class names and class indices match correctly in the combined dataset.

Based on the provided information, the `nc` (number of classes) in the YAML file is set to 83, but only a subset of classes is mentioned in the names field. It's possible that the missing classes from the existing dataset and the custom dataset are not properly mapped, leading to misclassifications during training and inference.

To resolve this issue, verify that the class names and indices from the existing dataset and the custom dataset are correctly aligned in the combined dataset's YAML file. Ensure that all classes from both datasets are included in the `names` field with accurate class indices.

Once the class mapping is verified and corrected, retrain the model using the combined dataset to recognize both the classes from the existing dataset and the custom dataset effectively.

Feel free to reach out if you need further assistance!
